Bottom: fa11b092fd700ef029320f26e302b370fbd43db4
Top:    e5cad329c8c5675b412e1196fe6477e3ff90a48f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 23:49:02 +0100

Refresh of draw-c-change-drawtop-to-drawo

---

diff --git a/src/draw.c b/src/draw.c
index 35810b3..4cac1d8 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -385,12 +385,13 @@ DrawEverything (BoxTypePtr drawn_area)
 	  if (DrawLayerGroup (group, drawn_area) && !gui->gui)
 	    {
 	      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
+	      r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
+	      /* draw element pads */
 	      if (group == component || group == solder)
 		{
                   side = (group == solder) ? SOLDER_LAYER : COMPONENT_LAYER;
 		  r_search (PCB->Data->pad_tree, drawn_area, NULL, pad_callback, &side);
 		}
-	      r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
 
 	      /* draw holes */
 	      plated = -1;
